### ISSUE TITLE
Turbolinks visit does not scroll to in-page anchors

### ISSUE THREAD

**Issue by bhox** [turbolinks/turbolinks#214]
I have a bunch of anchors with inline links and when clicked turbolinks will reload the page, fetching the full url as a visit instead of scrolling to the anchor. 

For example, a link to `example.com/page#potato` will load `example.com/page#potato` instead of simply scrolling to the inline anchor.

I fixed it by adding `data-turbolinks="false" `to all inline anchor links, but I don't see any reason to not do this by default. Links with only hash fragments are inline and should be ignored by turbolinks (unless I'm missing something!).

**Comment by benhoxie** [comment]
Similar problem when following a visit to a new page that includes an inline anchor via hash fragment. 

The page loads but does not scroll to a matching named element on page (the default html behavior).

**Comment by benhoxie** [comment]
dupe of: https://github.com/turbolinks/turbolinks/issues/75

**Comment by benhoxie** [comment]
Reopening for the second comment, following links with inline anchors. Only the first part is a dupe.

**Comment by benhoxie** [comment]
I used this to scroll to inline anchors if they are present. It's not fancy.

`
$(document).on('turbolinks:load',function(event){
  // scroll to matching named element in hash
  if(window.location.hash){
    tags = document.getElementsByName(window.location.hash.slice(1))
    if(tags.length > 0){
      window.scroll(0,tags[0].offsetTop);
    }
  }

});
`

**Comment by Tongboy** [comment]
Reading the cancelling visits section of the readme suggested a possible workaround - https://github.com/turbolinks/turbolinks#canceling-visits-before-they-start

In the `turbolinks:before-visit` event check if the url is the same as the current page and if so call `event.preventDefault()` to stop the visit and then scroll to the anchor.

thoughts?

**Comment by joeljackson** [comment]
It would be nice to have a systematic recommended approach to this.

This has worked for me, but it feels like every use case of link I introduce I need to think through as opposed to it just working.

```
document.addEventListener('turbolinks:before-visit', function(e) {
  var parser = document.createElement('a');
  parser.href = e.data.url;

  if(parser.host == window.location.host && parser.pathname == window.location.pathname) {
    e.preventDefault();
    window.location.href = e.data.url;
  }
});
```

This idea is if the host and path have not changed maybe there is no need to trigger turbolinks, just use the browser behavior.

**Comment by roooodcastro** [comment]
Neither of the workarounds presented here didn't work for me, Turbolinks is awesome but it's still very wrong when it comes to anchor links, and this issue seems to be present since 2016. There's a PR #285 to fix this but seems largely ignored by the owners.

**Comment by RaymondFallon** [comment]
I am seeing a very similar problem. When navigating to a different, cached page, it is doing the following:

1. Instantly loading the old page and (correctly) navigating to the anchor referenced in the link. 
2. Loading the page.
3. Navigating to the top of the page once the loading is complete.

NOTE: If the page is *not* cached, then the link is working as desired: loading the page and navigating down to the anchor mentioned at the end of the URL. 

This is fixed by turning turbolinks off on this link (as mentioned above), but it would be much nicer if this behavior worked *with* turbolinks, as expected.

**Comment by domchristie** [comment]
Scrolling to named anchors should be fixed in https://github.com/turbolinks/turbolinks/pull/348 and released in v5.1
Closing as a duplicate of #75
